src/relay/op/nn/convolution.cc

```diff
@@ -52,6 +50,67 @@ Array<Array<Layout> > ConvInferCorrectLayout(
                                    params->data_layout : params->out_layout}};
 }
 
+
+// relay.nn.conv1d
+TVM_REGISTER_NODE_TYPE(Conv1DAttrs);
+
+// Positional relay function to create conv1d operator
+// used by frontend FFI.
+Expr MakeConv1D(Expr data,
```

Comment:
Can you clean up MakeConv1D, MakeConv2D, and MakeConv3D using template like we do in pooling.cc?

Reply:
Good catch, you're right that it was very straightforward to combine all the MakeConv calls. My latest commit includes this change.